amundsen_application/static/js/ducks/allTags/tests/index.spec.ts

```diff
@@ -3,17 +3,37 @@ import * as matchers from 'redux-saga-test-plan/matchers';
 import { throwError } from 'redux-saga-test-plan/providers';
 
 import { metadataAllTags } from '../api/v0';
-import reducer, { getAllTags, initialState, AllTagsReducerState } from '../reducer';
+import reducer, {
+  getAllTags, getAllTagsFailure, getAllTagsSuccess,
+  initialState, AllTagsReducerState
+} from '../reducer';
 import { getAllTagsWatcher, getAllTagsWorker } from '../sagas';
 import { GetAllTags } from '../types';
 
 describe('allTags ducks', () => {
   describe('actions', () => {
-    describe('getAllTags', () => {
-      it('should return action of type GetAllTagsRequest', () => {
-        expect(getAllTags()).toEqual({ type: GetAllTags.REQUEST });
+    it('getAllTags - returns the action to get all tags', () => {
+      expect(getAllTags()).toEqual({ type: GetAllTags.REQUEST });
+    });
+
+    it('getAllTagsFailure - returns the action to process failure', () => {
+      expect(getAllTagsFailure()).toEqual({
+        type: GetAllTags.FAILURE,
+        payload: {
+          tags: [],
+        },
```

Comment:
I think I can get behind that. What are your thoughts on the perspective that unit tests should only have one assertion? This will end up being slightly more code to separate the assertions.

Reply:
I don't think we need to limit one assertion per block. I think they can be grouped in logical blocks according to the scenario/case that's being tested. For example you could test that an HTTP response is correct and group the related assertions together. 
```
... setup
expect(response.status).toEqual(200);
expect(response.data.tags).toEqual(expectedTags);
```
If either test fails the error log will tell you which particular assertion failed, even if they share the same test block. In contrast doing a `expect(complexObj).equals(expectedComplexObject)` you will get a large diff printed out which may be difficult to decipher.

Reply:
Also can get behind that reasoning, I will update the actions tests to assert on specific values that we expect to be on the action as opposed to the entire action itself.